domain/src/ValueObjects/Transaction.php

```diff
@@ -34,11 +38,14 @@ public function __construct(
         public readonly Quantity $exchangeFeeQuantity,
         public readonly ?FiatAmount $exchangeFeeCostBasis,
     ) {
+        $this->sentAsset = $sentAssetIsNft ? $sentAsset : AssetSymbolNormaliser::normalise($sentAsset);
+        $this->receivedAsset = $receivedAssetIsNft ? $receivedAsset : AssetSymbolNormaliser::normalise($receivedAsset);
+
         match ($operation) {
-            Operation::Receive => $this->validateReceive($sentAsset, $receivedAsset, $receivedQuantity),
-            Operation::Send => $this->validateSend($sentAsset, $sentQuantity, $receivedAsset),
-            Operation::Swap => $this->validateSwap($sentAsset, $sentQuantity, $receivedAsset, $receivedQuantity),
-            Operation::Transfer => $this->validateTransfer($sentAsset, $sentQuantity, $receivedAsset),
+            Operation::Receive => $this->validateReceive(),
+            Operation::Send => $this->validateSend(),
+            Operation::Swap => $this->validateSwap(),
+            Operation::Transfer => $this->validateTransfer(),
```

Comment:
ℹ️ There was no need to pass these values as parameters as they are already assigned to the object